### PR TITLE
Inject vs each_with_object in ruby

### PR DESCRIPTION
**Pull request by rShetty**
Comparison of the two 'foldl' functions in Ruby (inject and each_with_object).
Comparing their working and semantics.

### CODE REVIEW

**Comment by vishnun** [comment]
Thanks a lot for sharing. Each_with_object is more cleaner and more readable. Will save a lot of debugging time as well.

File: ruby/inject_vs_each_with_object.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+### Inject
```
Comment (maurogeorge) [inline ruby/inject_vs_each_with_object.md]: I think it is better to start with a h1 title like "Inject vs each_with_object" and here use the h2 title.

File: ruby/inject_vs_each_with_object.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  inject_array.inject({}) do |accumulator, value|
+    accumulator[value] = value.downcase
+    accumulator
+  end
+
+  Output: => {"A"=>"a", "B"=>"b", "C"=>"c", "D"=>"d"}
+  ```
+
+### each_with_object
```
Comment (maurogeorge) [inline ruby/inject_vs_each_with_object.md]: Use the h2 title.

File: ruby/inject_vs_each_with_object.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  array.each_with_object({}) do |value, accumulator|
+    accumulator[value] = value.downcase
+  end
+
+  Output: => {"A"=>"a", "B"=>"b", "C"=>"c", "D"=>"d"}
+  ```
+One more thing which you can notice is the order of arguments to the block for each of the functions.
+
+Inject takes the result/accumulator and then the iteratable value, whereas 'each_with_object' takes the value followed by the result/accumulator.
+
```
Comment (maurogeorge) [inline ruby/inject_vs_each_with_object.md]: Remove this whitespaces.

**Comment by maurogeorge** [comment]
@rShetty good TIL :+1:

**Comment by rShetty** [comment]
@maurogeorge  Made changes based on some of your feedback. Thanks for the feedback.

**Comment by JoelQ** [comment]
I've merged this as https://github.com/thoughtbot/til/commit/4c2b65c49e5d513a2e5ecb582f80931eacbf44b3, thanks for sharing!

**Pull request closed without merging**